Improve readme (#12)
teamdigitale/anpr-dashboard-server: README.md

# ANPR Dashboard Server

The ANPR dashboard server is a go-based application that provides data related to the migration status of Italian municipalities to the National Registry.

> The service exposes data both through an API interface and through a UI, through which it's possible to both download existing datasets and upload new data in CSV format.

Reviewer: ```suggestion
The service exposes data both via API and via an UI, that you can use to both download existing datasets and upload new data in CSV format.
```

## Main components

Some of the folders in this repository are particularly significant:

* **converter**: a tool to convert the data (schede) collected by [SOGEI](http://www.sogei.it/) into json files, by adding geolocation informations from Google maps.

* **site**: the website that shows the main statistical data, that allow users to download a csv file with the latest data and to upload a csv with new data

* **openapi**: the OpenAPI 3 specification of the APIs exposed

## GUI/dashboard

> The application also exposes a web-based GUI that can be accessed from website root ('/') once the server is running.

Reviewer: ```suggestion
The application also exposes a web UI that can be accessed from website root ('/') once the server is running.
```

### Run the project directly on the local machine

```shell
# Configuration
cd anpr-dashboard-server/server
mkdir creds
cp {THREE-CONFIG-FROM-ADMINS} creds
mkdir dashboard-subentro
cp {ANOTHER-CONFIG-FILE-FROM-ADMINS} subentroconfig.yaml dashboard-subentro/

# Build the project
make build

# Run the service
./server --mode=debug --config-file=/<PATH-TO-YOUR-WORKSPACE>/anpr-dashboard-server/dashboard-subentro/subentroconfig.yaml
```

### Run the project as a Docker container

A `Dockerfile` and a `docker-compose.yaml` files are in the root of this repository.

Create a new *vault* directory in the root of this repository. Put your development configurations and credentials in it. Make sure they reference */srv/db/sqlite.db*. Put the sqlite database in *vault/db* and name it *sqlite.db*

To build the local test environment run:

```shell
UID=$UID docker-compose up -d
```

The website and the APIs should now be accessible on port *8080*. While the website is accessible at the root, the APIs can be accessed under the path */api*

To bring down the test environment and remove the containers use

```shell
docker-compose down
```

> ## Query examples

Reviewer: @mccalv le query possiamo metterle in un altro file? Ha senso?

Author: @mccalv @ioggstream ok, le metto in un file a parte e faccio un link nel readme principale
